lib/core/util.js

```diff
@@ -26,6 +26,60 @@ function isBlobLike (object) {
   )
 }
 
+function isDate (val) {
+  return toString.call(val) === '[object Date]'
+}
+
+function isObject (val) {
+  return val !== null && typeof val === 'object'
+}
+
+// based on https://github.com/axios/axios/blob/63e559fa609c40a0a460ae5d5a18c3470ffc6c9e/lib/helpers/buildURL.js (MIT license)
+// this escapes all non-browser-url friendly characters, but restores the ones that are supported by browsers
+function encode (val) {
+  return encodeURIComponent(val).replace(/%3A/gi, ':').replace(/%24/g, '$').replace(/%2C/gi, ',').replace(/%20/g, '+').replace(/%5B/gi, '[').replace(/%5D/gi, ']')
+}
```

Comment:
Why do we need this?

> ll non-browser-url friendly characters, but restores the ones that are supported by browsers

We are not in a browser?

Reply:
Something glitched in my head, sorry. I think restoring non-escaped symbols was purely to conserve bandwidth, and I'm OK to just use encodeURIComponent for everything. Will do the change.